auth_jwt/tests/test_auth_jwt.py

```diff
@@ -0,0 +1,250 @@
+# Copyright 2021 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import contextlib
+import time
+from unittest.mock import Mock
+
+import jwt
+
+import odoo.http
+from odoo.exceptions import ValidationError
+from odoo.tests.common import TransactionCase
+from odoo.tools import mute_logger
+from odoo.tools.misc import DotDict
+
+from ..exceptions import (
+    AmbiguousJwtValidator,
+    JwtValidatorNotFound,
+    UnauthorizedInvalidToken,
+    UnauthorizedMalformedAuthorizationHeader,
+    UnauthorizedMissingAuthorizationHeader,
+    UnauthorizedPartnerNotFound,
+)
+
+
+class TestAuthMethod(TransactionCase):
+    @contextlib.contextmanager
+    def _mock_request(self, authorization):
```

Comment:
you know that you can use `website.tools.MockRequest`?

Reply:
Yea, but it does not allow mocking http headers, so I copied the part I need.
Also, that MockRequest should be in odoo.tests.common :)

Reply:
I know, I hate it but you can use it this way https://github.com/shopinvader/odoo-shopinvader/blob/416b45d7b02b163550e40b03bb64c61a41237d05/shopinvader_customer_multi_user/tests/test_multi_user_service_ctx.py#L18

Reply:
not a blocker of course ;)

Reply:
Yes, sometimes I prefer a little bit of copy/paste over hacking over undocumented parts of Odoo (sic!).